python/lsst/analysis/tools/analysisMetrics/analysisMetrics.py

```diff
@@ -0,0 +1,278 @@
+
+        self.process.buildActions.mags = MagColumnNanoJansky(vectorKey="{band}_psfFlux")
+        self.process.buildActions.fracDiff = FractionalDifference(
+            actionA=CalcShapeSize(),
+            actionB=CalcShapeSize(colXx="{band}_ixxPSF", colYy="{band}_iyyPSF", colXy="{band}_ixyPSF"),
+        )
+        # pre-compute a stellar selector mask so it can be used in the filter
+        # actions while only being computed once, alternatively the stellar
+        # selector could be calculated and applied twice in the filter stage
+        self.process.buildActions.starSelector = StellarSelector()
+
+        self.process.filterActions.xStars = DownselectVector(
+            vectorKey="mags", selector=VectorSelector(vectorKey="starSelector")
+        )
+        self.process.filterActions.yStars = DownselectVector(
+            vectorKey="fracDiff", selector=VectorSelector(vectorKey="starSelector")
+        )
+        # downselect the psfFlux as well
+        self.process.filterActions.psfFlux = DownselectVector(
+            vectorKey="{band}_psfFlux", selector=VectorSelector(vectorKey="starSelector")
+        )
+        self.process.filterActions.psfFluxErr = DownselectVector(
+            vectorKey="{band}_psfFluxErr", selector=VectorSelector(vectorKey="starSelector")
+        )
+
+        self.process.calculateActions.stars = ScatterPlotStatsAction(
+            vectorKey="yStars",
+        )
+        # use the downselected psfFlux
+        self.process.calculateActions.stars.highSNSelector.fluxType = "psfFlux"
+        self.process.calculateActions.stars.lowSNSelector.fluxType = "psfFlux"
+        self.process.calculateActions.stars.fluxType = "psfFlux"
+
+        self.post_process.units = {  # type: ignore
+            "{band}_highSNStars_median": "pixel",
+            "{band}_highSNStars_sigmaMad": "pixel",
+            "{band}_highSNStars_count": "count",
+            "{band}_lowSNStars_median": "pixel",
+            "{band}_lowSNStars_sigmaMad": "pixel",
+            "{band}_lowSNStars_count": "count",
+        }
+
+
+class StellarLocusBaseMetric(AnalysisMetric):
+    # Use this as the Base Class for now StellarLocusBaseMetric
+    parameterizedBand: bool = False
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.prep.selectors.flagSelector = CoaddPlotFlagSelector()
+
+        self.prep.selectors.snSelector = SnSelector()
+        self.prep.selectors.snSelector.threshold = 300
+
+        self.prep.selectors.starSelector = StellarSelector()
+
+        self.process.buildActions.x = ExtinctionCorrectedMagDiff()
+        self.process.buildActions.x.magDiff.returnMillimags = False
+        self.process.buildActions.y = ExtinctionCorrectedMagDiff()
+        self.process.buildActions.y.magDiff.returnMillimags = False
+
+
+class GRIStellarPSFMetric(AnalysisMetric):
+    # Use this as the Base Class for now StellarLocusBaseMetric
```

Comment:
This comment seems inconsistent with the class inheritance pattern that is being used.